src/framework/script/script.js

```diff
@@ -65,16 +64,13 @@ function createScript(name, app) {
     if (reservedScriptNames.has(name))
         throw new Error(`Script name '${name}' is reserved, please rename the script`);
 
-    const scriptType = function (args) {
-        EventHandler.prototype.initEventHandler.call(this);
-        ScriptType.prototype.initScriptType.call(this, args);
-    };
+    class scriptType extends ScriptType {}
```

Comment:
What about something like:

```
    class ScriptTypeWithAttributes extends ScriptType {
        attributes = new ScriptAttributes(this);
    }

    registerScript(ScriptTypeWithAttributes, name, app);
    return ScriptTypeWithAttributes;
    ```

Reply:
I did try that but I got a linting error for using `this` as a parameter. Seems to work fine tho if I just ignore the linting error

Reply:
Actually so does the existing method - The linting just didnt show before 👍